File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/components/StepperProgressFragment.java
package com.nasahapps.materialdesigntoolbox.example.ui.components;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.view.View;

import com.nasahapps.materialdesigntoolbox.components.OnStepperProgressListener;
import com.nasahapps.materialdesigntoolbox.components.StepperProgressLayout;
import com.nasahapps.materialdesigntoolbox.example.R;
import com.nasahapps.materialdesigntoolbox.example.ui.adapter.MockViewPagerAdapter;
import com.nasahapps.materialdesigntoolbox.example.ui.main.MainActivity;

import butterknife.Bind;

/**
 * Created by Hakeem on 4/17/16.
 */
public class StepperProgressFragment extends ComponentFragment {

    private static final String EXTRA_TYPE = "type";

    @Bind(R.id.viewPager)
    ViewPager mViewPager;
    @Bind(R.id.layout)
    StepperProgressLayout mLayout;

    public static StepperProgressFragment newInstance(@StepperProgressLayout.ProgressType int type) {
        Bundle args = new Bundle();
        args.putInt(EXTRA_TYPE, type);

        StepperProgressFragment fragment = new StepperProgressFragment();
        fragment.setArguments(args);

        return fragment;
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_stepper_progress;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mViewPager.setAdapter(new MockViewPagerAdapter(getFragmentManager()));
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                mLayout.setProgress(position + 1);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        mLayout.setMaxProgress(MockViewPagerAdapter.MAX_COUNT);
        mLayout.setProgressType(getArguments().getInt(EXTRA_TYPE));
        mLayout.addOnStepperProgressListener(new OnStepperProgressListener() {
            @Override
            public void onStepSelected(int position) {
                mViewPager.setCurrentItem(position, true);
            }

            @Override
            public void onStepDeselected(int position) {

            }

            @Override
            public void onStepCompleted(int position) {

            }

            @Override
            public void onStepsFinished() {

            }
        });
    }

    @Override
    public void onResume() {
        super.onResume();
        ((MainActivity) getActivity()).setToolbarTitle("Steppers");
    }
}

File: app/src/main/java/com/nasahapps/mdt/example/ui/components/StepperProgressFragment.kt
package com.nasahapps.mdt.example.ui.components

import android.os.Bundle
import android.support.v4.view.ViewPager
import android.view.View
import com.nasahapps.mdt.components.StepperProgressLayout
import com.nasahapps.mdt.example.R
import com.nasahapps.mdt.example.ui.main.MainActivity
import com.nasahapps.mdt.example.ui.mock.MockViewPagerAdapter
import kotlinx.android.synthetic.main.fragment_stepper_progress.*

/**
 * Created by Hakeem on 4/17/16.
 */
class StepperProgressFragment : ComponentFragment() {

    override fun getLayoutId(): Int {
        return R.layout.fragment_stepper_progress
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewPager?.adapter = MockViewPagerAdapter(fragmentManager)
        viewPager?.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                layout?.progress = position + 1
            }

            override fun onPageScrollStateChanged(state: Int) {

            }
        })
        layout?.maxProgress = MockViewPagerAdapter.MAX_COUNT
        layout?.progressType = arguments.getInt(EXTRA_TYPE)
        layout?.addOnStepperProgressListener(object : StepperProgressLayout.OnStepperProgressListener {
            override fun onStepSelected(position: Int) {
                viewPager?.setCurrentItem(position, true)
            }

            override fun onStepDeselected(position: Int) {

            }

            override fun onStepsFinished() {

            }
        })
    }

    override fun onResume() {
        super.onResume()
        (activity as MainActivity).setToolbarTitle("Steppers")
    }

    companion object {

        private val EXTRA_TYPE = "type"

        fun newInstance(@StepperProgressLayout.ProgressType type: Int): StepperProgressFragment {
            val args = Bundle()
            args.putInt(EXTRA_TYPE, type)

            val fragment = StepperProgressFragment()
            fragment.arguments = args

            return fragment
        }
    }
}
